src/database.py

```diff
@@ -44,7 +45,64 @@
 )
 from utils import filter_forecast_values, floor_30_minutes_dt, get_start_datetime
 
-db_conn = DatabaseConnection(url=os.getenv("DB_URL", "not_set"))
+
+class BaseDBClient(abc.ABC):
+    """This is a base class for database clients with one static method get_client().
+
+    Methods
+    -------
+    get_client : static method
+        It gets the database client. If a valid Postgresql database URL is set in
+        the "DB_URL" environment variable, get_client returns an instance
+        of DatabaseConnection(). If not, it returns an instance of the DummyDBClient.
+    """
+
+    @staticmethod
+    def get_client():
```

Comment:
could we rename name this to `get_connection`, sqlalmacy, seems to ahev `connections` and `clients` so we should try and stick with those

Reply:
Yep, sorry this is a little confused over what is a client and what is a connection... it's basically both atm 😅 have changed